### PR TITLE
[DONT MERGE] WIP

### PR DESCRIPTION
**Pull request by ahmetmircik**

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheImpl.java [hunk 1]
```diff
@@ -248,6 +241,41 @@ public boolean remove(Data key) {
     }
 
     @Override
+    public Object mapKeyToMarker(Data key) {
+        NearCacheRecord markerRecord = new NearCacheRecord(key, MARKER_VALUE);
+        NearCacheRecord previous = cache.put(key, markerRecord);
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheImpl.java]: shouldn't this be CAS too? what if some other thread managed to populate the near cache? then you will remove completely good value.

**Comment by ahmetmircik** [comment]
fix will be handled differently later, no need this PR now.

**Pull request closed without merging**